Solutions/CohesitySecurity/Data Connectors/Helios2Sentinel/IncidentProducer/IncidentProducer.cs

```diff
@@ -0,0 +1,263 @@
+            }
+        }
+
+        private static string GetData(string path)
+        {
+            if (CloudStorageAccount.TryParse(blobStorageConnectionString, out var storageAccount))
+            {
+                // Create the CloudBlobClient that represents the Blob storage endpoint for the storage account.
+                var cloudBlobClient = storageAccount.CreateCloudBlobClient();
+                var container = cloudBlobClient.GetContainerReference(containerName);
+                var blobRef = container.GetBlockBlobReference(path);
+
+                using (var ms = new MemoryStream())
+                    using (var sr = new StreamReader(ms))
+                    {
+                        var task = blobRef.DownloadToStreamAsync(ms);
+                        task.Wait();
+                        ms.Position = 0;
+                        return sr.ReadToEnd();
+                    }
+            }
+            else
+            {
+                throw new InvalidProgramException("Invalid format string");
+            }
+        }
+
+        private static void WriteData(string path, string data, ILogger log)
+        {
+            try
+            {
+                if (CloudStorageAccount.TryParse(blobStorageConnectionString, out var storageAccount))
+                {
+                    var cloudBlobClient = storageAccount.CreateCloudBlobClient();
+                    var container = cloudBlobClient.GetContainerReference(containerName);
+                    var blobRef = container.GetBlockBlobReference(path);
+                    var dataAsBytes = Encoding.UTF8.GetBytes(data);
+                    blobRef.UploadFromByteArrayAsync(dataAsBytes, 0, dataAsBytes.Length).Wait();
+                }
+                else
+                {
+                    throw new InvalidProgramException("Invalid format string");
+                }
+            }
+            catch (Exception ex)
+            {
+                log.LogError("Exception --> " + ex.Message);
+            }
+        }
+
+        private static void TextWriter(string incidentID, string param, string value, ILogger log)
+        {
+            WriteData(incidentID + "\\" + param, value, log);
+        }
+
+        [FunctionName("IncidentProducer")]
+        public static async Task RunAsync(
+#if DEBUG
+            [TimerTrigger("*/30 * * * * *")] TimerInfo myTimer,
+#else
+            [TimerTrigger("* */5 * * * *")]TimerInfo myTimer,
+#endif
+            [Queue("%CohesityQueueName%"), StorageAccount("AzureWebJobsStorage")] ICollector<string> outputQueueItem,
+            ILogger log)
+        {
+            log.LogInformation($"C# Timer trigger function executed at: {DateTime.Now}");
+            long startDateUsecs = 0;
+
+            try
+            {
+                var db = Connection.GetDatabase();
+                string apiKey = GetSecret("ApiKey");
+                string blobKey = GetSecret("workspace") + "\\" + apiKey;
```

Comment:
You can keep workspace in the env variable. Workspace name is not a secret. Users may also may want to tweak to redirect incidents to another workspace.